saw-core/src/SAWCore/SharedTerm.hs

```diff
@@ -2773,18 +2788,18 @@ scInstantiateExt :: SharedContext -> IntMap Term -> Term -> IO Term
 scInstantiateExt sc vmap
   | all termIsClosed vmap = scInstantiateExtClosed sc vmap
   | otherwise = instantiateVars sc fn 0
-  where fn _rec l (Left ec) =
+  where fn r l (Left ec) =
             case IntMap.lookup (ecVarIndex ec) vmap of
                Just t  -> incVars sc 0 l t
-               Nothing -> scVariable sc ec
+               Nothing -> scVariable sc =<< traverse r ec
         fn _ _ (Right i) = scLocalVar sc i
 
 -- | Internal variant of 'scInstantiateExt' that requires the
 -- substituted terms to all be closed, i.e. they must not have any
 -- loose de Bruijn indices.
 scInstantiateExtClosed :: SharedContext -> IntMap Term -> Term -> IO Term
 scInstantiateExtClosed sc vmap t0 =
-  do let vs = IntMap.keysSet vmap
+  do -- let vs = IntMap.keysSet vmap
```

Comment:
Delete the `-- let vs = IntMap.keysSet vmap` comment here? (And similarly for the `-- - | IntSet.disjoint vs (freeVars t) = pure t` comment below.)